### PR TITLE
[IOTDB-104]fix that MManager is incorrectly recovered when system reboots

### PR DESCRIPTION
**Pull request by jt2594838**
Please see https://issues.apache.org/jira/browse/IOTDB-104 for details

### CODE REVIEW

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java [hunk 1]
```diff
@@ -897,6 +899,29 @@ public void deleteOneFileNode(String processorName) throws FileNodeManagerExcept
...
+    // free and close the streams under this overflow directory
+    if (!overflowDir.exists()) {
+      return;
+    }
+    File[] overflowSubDirs = overflowDir.listFiles();
+    if (overflowSubDirs == null) {
+      return;
+    }
+    for (File overflowSubDir : overflowSubDirs) {
+      File[] overflowFiles = overflowSubDir.listFiles();
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: line 915-920 are duplicated with 934-939

File: iotdb/src/main/java/org/apache/iotdb/db/metadata/MGraph.java [hunk 2]
```diff
@@ -89,7 +89,7 @@ public void addPathToMTree(String path, TSDataType dataType, TSEncoding encoding
   /**
    * Add a seriesPath to {@code PTree}.
    */
-  public void addPathToPTree(String path) throws PathErrorException, MetadataArgsErrorException {
+  public void addPathToPTree(String path) throws PathErrorException, MetadataErrorException {
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/metadata/MGraph.java]: No ", MetadataErrorException" exception

File: iotdb/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 3]
```diff
@@ -298,13 +363,105 @@ public void addPathToMTree(String path, String dataType, String encoding)
...
+      throws MetadataErrorException {
+    return deletePathFromMTree(new Path(deletePath));
+  }
+
+  public Pair<Set<String>, Set<String>> deletePathFromMTree(Path deletePath)
+      throws MetadataErrorException {
+    return deletePathsFromMTree(Collections.singletonList(deletePath));
+  }
+
+  public Pair<Set<String>, Set<String>> deletePathsFromMTree(List<Path> deletePathList)
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: what is the meaning of the returned value?

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java (hunk 1 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java]: Is a duplication of 5 lines intolerable?

File: iotdb/src/main/java/org/apache/iotdb/db/metadata/MGraph.java (hunk 2 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/metadata/MGraph.java]: Thanks for the correction.

**Comment by jt2594838** [comment]
I think you have missed something. The Path.getMeasurment() is called in some situations, so you cannot avoid splitting the origin path. Wrapping with a Path does not make anything more complicated.

File: iotdb/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 3 above)
Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: As the names of the returned variable suggest, StorageGroups to be closed and StorageGroups to be deleted.

Comment (jixuan1989) [reply iotdb/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: javadoc needed...

Comment (jt2594838) [reply iotdb/src/main/java/org/apache/iotdb/db/metadata/MManager.java]: added

**Pull request merged**